### DISCUSSION TITLE
build failed on Rocky 8.5

### DISCUSSION THREAD

**Discussion by perlang** [AFLplusplus/AFLplusplus#1306] [Q&A]
Hi,
I am a beginner of the AFL/AFL++. Sorry for asking for help if here is not the right place!

I want to build AFLplusplus on Rocky 8.5 with clang/llvm version 12.0 from rocky's repo.
but the "make distrib" could not pass.

Thanks in advanced !

Best Regards,

[sun@Rocky git]$ cd AFLplusplus/
[sun@Rocky AFLplusplus]$ ls
afl-cmin               afl-wine-trace   coresight_mode    frida_mode              LICENSE    test          utils
afl-cmin.bash          Android.bp       custom_mutators   GNUmakefile             Makefile   testcases
afl-persistent-config  Changelog.md     dictionaries      GNUmakefile.gcc_plugin  nyx_mode   test-instr.c
afl-plot               CITATION.cff     Dockerfile        GNUmakefile.llvm        qemu_mode  TODO.md
afl-system-config      config.h         docs              include                 README.md  types.h
afl-whatsup            CONTRIBUTING.md  dynamic_list.txt  instrumentation         src        unicorn_mode
[sun@Rocky AFLplusplus]$ make distrib
[*] Compiling afl++ for OS Linux on ARCH x86_64
[*] Checking for the default compiler cc...
[*] Testing the PATH environment variable...
[*] Checking for the ability to compile x86 code...
[+] shmat seems to be working.
[+] Python 3.10.2 support seems to be working.
[+] Everything seems to be working, ready to compile.
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2 -flto=full -c src/afl-common.c -o src/afl-common.o
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2 -flto=full -c src/afl-sharedmem.c -o src/afl-sharedmem.o
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2 -flto=full -c src/afl-forkserver.c -o src/afl-forkserver.o
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2 -Iinclude -fno-move-loop-invariants -fdisable-tree-cunrolli -O3 -fno-unroll-loops -c src/afl-performance.c -o src/afl-performance.o
cc1: note: disable pass tree-cunrolli for functions in the range of [0, 4294967295]
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2  -flto=full src/afl-fuzz-redqueen.c src/afl-fuzz-statsd.c src/afl-fuzz-extras.c src/afl-fuzz-queue.c src/afl-fuzz-cmplog.c src/afl-fuzz-state.c src/afl-fuzz-one.c src/afl-fuzz-python.c src/afl-fuzz-mutators.c src/afl-fuzz-stats.c src/afl-fuzz.c src/afl-fuzz-init.c src/afl-fuzz-bitmap.c src/afl-fuzz-run.c src/afl-common.o src/afl-sharedmem.o src/afl-forkserver.o src/afl-performance.o -o afl-fuzz -DUSE_PYTHON -I/usr/local/include/python3.10 -I/usr/local/include/python3.10 -ldl -lrt -lm -lpython3.10 -lcrypt -lpthread -ldl  -lutil -lm -lm  -L/usr/local/lib/python3.10/config-3.10-x86_64-linux-gnu -L/usr/local/lib -lpython3.10 -lcrypt -lpthread -ldl  -lutil -lm -lm  -DPYTHON_VERSION="\"Python 3.10.2\"" -ldl -lrt -lm -lm
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2  -flto=full src/afl-showmap.c src/afl-common.o src/afl-sharedmem.o src/afl-forkserver.o src/afl-performance.o -o afl-showmap -ldl -lrt -lm
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2  -flto=full src/afl-tmin.c src/afl-common.o src/afl-sharedmem.o src/afl-forkserver.o src/afl-performance.o -o afl-tmin -ldl -lrt -lm
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2  -flto=full src/afl-gotcpu.c src/afl-common.o -o afl-gotcpu -ldl -lrt -lm
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2  -flto=full src/afl-analyze.c src/afl-common.o src/afl-sharedmem.o src/afl-performance.o src/afl-forkserver.o -o afl-analyze -ldl -lrt -lm
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -g -Wno-pointer-sign -Wno-variadic-macros -Wall -Wextra -Wpointer-arith -I include/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DDOC_PATH=\"/usr/local/share/doc/afl\" -D_FORTIFY_SOURCE=2 src/afl-as.c -o afl-as -ldl -lrt -lm
make -j4 -f GNUmakefile.llvm
make[1]: Entering directory '/home/sun/git/AFLplusplus'
[+] llvm_mode detected llvm 10+, enabling neverZero implementation and c++14
[+] llvm_mode detected llvm 11+, enabling afl-lto LTO implementation
GNUmakefile.llvm:218: ld.lld not found, cannot enable LTO mode
[+] shmat seems to be working.
[*] Checking for working 'llvm-config'...
[*] Checking for working '/usr/bin/clang'...
[*] Checking for matching versions of '/usr/bin/clang' and 'llvm-config'
[*] We have llvm-config version 12.0.1 with a clang version 12.0.1, good.
[*] Checking for './afl-showmap'...
[+] All set and ready to build.
/usr/bin/clang -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  -c src/afl-common.c -o instrumentation/afl-common.o 
In file included from src/afl-common.c:35:
In file included from ./include/debug.h:28:
In file included from ./include/types.h:28:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:5:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang -I/usr/include -I./../include -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  src/afl-cc.c instrumentation/afl-common.o -o afl-cc -DLLVM_MINOR=1 -DLLVM_MAJOR=12  -DCFLAGS_OPT=\"\" -lm
In file included from src/afl-cc.c:20:
In file included from ./include/common.h:35:
In file included from ./include/types.h:28:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:5:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang -I/usr/include -I./../include -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  -O3 -Wno-unused-result -fPIC -c instrumentation/afl-compiler-rt.o.c -o afl-compiler-rt.o
In file included from instrumentation/afl-compiler-rt.o.c:20:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:4:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
[*] Building 32-bit variant of the runtime (-m32)... failed (that's fine)
[*] Building 64-bit variant of the runtime (-m64)... success!
/usr/bin/clang++ -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  `llvm-config --cxxflags` -fno-rtti -fPIC -std=c++14 -c instrumentation/afl-llvm-common.cc -o instrumentation/afl-llvm-common.o 
In file included from instrumentation/afl-llvm-common.cc:3:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:5:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -Wdeprecated -fno-rtti -fPIC -std=c++14 -shared instrumentation/afl-llvm-pass.so.cc -o afl-llvm-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
In file included from instrumentation/afl-llvm-pass.so.cc:31:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:6:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -Wdeprecated -fno-rtti -fPIC -std=c++14 -shared instrumentation/SanitizerCoveragePCGUARD.so.cc -o SanitizerCoveragePCGUARD.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
In file included from instrumentation/SanitizerCoveragePCGUARD.so.cc:48:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:6:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/split-compares-pass.so.cc -o split-compares-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/split-switches-pass.so.cc -o split-switches-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/cmplog-routines-pass.cc -o cmplog-routines-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/cmplog-instructions-pass.cc -o cmplog-instructions-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/cmplog-switches-pass.cc -o cmplog-switches-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/afl-llvm-dict2file.so.cc -o afl-llvm-dict2file.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
In file included from instrumentation/afl-llvm-dict2file.so.cc:21:
./include/config.h:29:9: warning: 'VERSION' macro redefined [-Wmacro-redefined]
#define VERSION "++3.15a"
        ^
<command line>:6:9: note: previous definition is here
#define VERSION ""
        ^
1 warning generated.
/usr/bin/clang++ `llvm-config --cxxflags` -fno-rtti -fPIC -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -I ./include/ -DVERSION=\"\" -Wno-variadic-macros -DLLVM_MINOR=1 -DLLVM_MAJOR=12 -Wno-deprecated-declarations -Wl,-znodelete -shared instrumentation/compare-transform-pass.so.cc -o compare-transform-pass.so `llvm-config --ldflags`  instrumentation/afl-llvm-common.o
[*] Testing the CC wrapper and instrumentation output...
unset AFL_USE_ASAN AFL_USE_MSAN AFL_INST_RATIO; ASAN_OPTIONS=detect_leaks=0 AFL_QUIET=1 AFL_PATH=. AFL_LLVM_LAF_ALL=1 ./afl-cc -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  ./test-instr.c -o test-instr 
ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr0 ./test-instr < /dev/null
echo 1 | ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr1 ./test-instr
[+] All right, the instrumentation seems to be working!
[+] All done! You can now use './afl-cc' to compile programs.
make[1]: Leaving directory '/home/sun/git/AFLplusplus'
make -f GNUmakefile.gcc_plugin
make[1]: Entering directory '/home/sun/git/AFLplusplus'
[+] shmat seems to be working.
[*] Checking for working 'cc'...
[*] Checking for gcc plugin development header files...
[*] Checking for './afl-showmap'...
[+] All set and ready to build.
g++ -O3 -g -funroll-loops -D_FORTIFY_SOURCE=2 -Wall -std=c++11 -fPIC -fno-rtti -I"/usr/lib/gcc/x86_64-redhat-linux/8/plugin"/include -I"/usr/lib/gcc/x86_64-redhat-linux/8/plugin" -shared instrumentation/afl-gcc-pass.so.cc -o afl-gcc-pass.so
ln -sf afl-cc afl-gcc-fast
ln -sf afl-cc afl-g++-fast
ln -sf afl-cc.8 afl-gcc-fast.8
ln -sf afl-cc.8 afl-g++-fast.8
[*] Building 32-bit variant of the runtime (-m32)... failed (that's fine)
[*] Testing the CC wrapper and instrumentation output...
unset AFL_USE_ASAN AFL_USE_MSAN; ASAN_OPTIONS=detect_leaks=0 AFL_QUIET=1 AFL_INST_RATIO=100 AFL_PATH=. AFL_CC=cc ./afl-gcc-fast -O3 -g -funroll-loops -D_FORTIFY_SOURCE=2 -Wall -Iinclude -Wno-pointer-sign -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DGCC_VERSION=\"\" -DGCC_BINDIR=\"\" -Wno-unused-function  ./test-instr.c -o test-instr -lrt
ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr0 ./test-instr </dev/null
echo 1 | ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr1 ./test-instr
[+] All right, the instrumentation seems to be working!
[+] All done! You can now use './afl-gcc-fast' to compile programs.
make[1]: Leaving directory '/home/sun/git/AFLplusplus'
[*] Testing the CC wrapper afl-cc and its instrumentation output...
afl-cc++3.15a by Michal Zalewski, Laszlo Szekeres, Marc Heuse - mode: LLVM-PCGUARD
SanitizerCoveragePCGUARD++3.15a
[+] Instrumented 11 locations with no collisions (non-hardened mode) of which are 0 handled and 0 unhandled selects.
ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr0 ./test-instr < /dev/null
echo 1 | ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr1 ./test-instr

[+] All right, the instrumentation of afl-cc seems to be working!
[+] Main compiler 'afl-cc' successfully built!
[+] LLVM mode for 'afl-cc' successfully built!
[-] LLVM LTO mode for 'afl-cc'  failed to build, this would need LLVM 11+, see instrumentation/README.lto.md how to build it
[+] gcc_plugin for 'afl-cc' successfully built!
[+] All done! Be sure to review the README.md - it's pretty short and useful.
NOTE: If you can read this, your terminal probably uses white background.
This will make the UI hard to read. See docs/status_screen.md for advice.
make -C utils/aflpp_driver
make[1]: Entering directory '/home/sun/git/AFLplusplus/utils/aflpp_driver'
/usr/bin/clang -I. -I../../include -O3 -funroll-loops -g -fPIC -c aflpp_driver.c
'libAFLDriver.a' -> '../../libAFLDriver.a'
/usr/bin/clang -O3 -funroll-loops -g -fPIC -O0 -funroll-loops -c aflpp_qemu_driver.c
'libAFLQemuDriver.a' -> '../../libAFLQemuDriver.a'
Note: Optional aflpp_qemu_driver_hook.o not built.
Note: Optional aflpp_qemu_driver_hook.so not built.
make[1]: Leaving directory '/home/sun/git/AFLplusplus/utils/aflpp_driver'
make -j4 -f GNUmakefile.llvm
make[1]: Entering directory '/home/sun/git/AFLplusplus'
[+] llvm_mode detected llvm 10+, enabling neverZero implementation and c++14
[+] llvm_mode detected llvm 11+, enabling afl-lto LTO implementation
GNUmakefile.llvm:218: ld.lld not found, cannot enable LTO mode
[+] shmat seems to be working.
[*] Checking for working 'llvm-config'...
[*] Checking for working '/usr/bin/clang'...
[*] Checking for matching versions of '/usr/bin/clang' and 'llvm-config'
[*] We have llvm-config version 12.0.1 with a clang version 12.0.1, good.
[*] Checking for './afl-showmap'...
[+] All set and ready to build.
[*] Building 32-bit variant of the runtime (-m32)... failed (that's fine)
[*] Testing the CC wrapper and instrumentation output...
unset AFL_USE_ASAN AFL_USE_MSAN AFL_INST_RATIO; ASAN_OPTIONS=detect_leaks=0 AFL_QUIET=1 AFL_PATH=. AFL_LLVM_LAF_ALL=1 ./afl-cc -O3 -funroll-loops -fPIC -D_FORTIFY_SOURCE=2 -Wall -g -Wno-cast-qual -Wno-variadic-macros -Wno-pointer-sign -I ./include/ -I ./instrumentation/ -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DLLVM_BINDIR=\"/usr/bin\" -DVERSION=\"\" -DLLVM_LIBDIR=\"/usr/lib64\" -DLLVM_VERSION=\"12.0.1\" -Wno-deprecated -DAFL_CLANG_FLTO=\"-flto=full\" -DAFL_REAL_LD=\"\" -DAFL_CLANG_LDPATH=\"\" -DAFL_CLANG_FUSELD=\"\" -DCLANG_BIN=\"/usr/bin/clang\" -DCLANGPP_BIN=\"/usr/bin/clang++\" -DUSE_BINDIR=1 -Wno-unused-function -fdebug-prefix-map="/home/sun/git/AFLplusplus=llvm_mode"  ./test-instr.c -o test-instr 
ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr0 ./test-instr < /dev/null
echo 1 | ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr1 ./test-instr
[+] All right, the instrumentation seems to be working!
[+] All done! You can now use './afl-cc' to compile programs.
make[1]: Leaving directory '/home/sun/git/AFLplusplus'
make -f GNUmakefile.gcc_plugin
make[1]: Entering directory '/home/sun/git/AFLplusplus'
[+] shmat seems to be working.
[*] Checking for working 'cc'...
[*] Checking for gcc plugin development header files...
[*] Checking for './afl-showmap'...
[+] All set and ready to build.
[*] Building 32-bit variant of the runtime (-m32)... failed (that's fine)
[*] Testing the CC wrapper and instrumentation output...
unset AFL_USE_ASAN AFL_USE_MSAN; ASAN_OPTIONS=detect_leaks=0 AFL_QUIET=1 AFL_INST_RATIO=100 AFL_PATH=. AFL_CC=cc ./afl-gcc-fast -O3 -g -funroll-loops -D_FORTIFY_SOURCE=2 -Wall -Iinclude -Wno-pointer-sign -DAFL_PATH=\"/usr/local/lib/afl\" -DBIN_PATH=\"/usr/local/bin\" -DGCC_VERSION=\"\" -DGCC_BINDIR=\"\" -Wno-unused-function  ./test-instr.c -o test-instr -lrt
ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr0 ./test-instr </dev/null
echo 1 | ASAN_OPTIONS=detect_leaks=0 ./afl-showmap -m none -q -o .test-instr1 ./test-instr
[+] All right, the instrumentation seems to be working!
[+] All done! You can now use './afl-gcc-fast' to compile programs.
make[1]: Leaving directory '/home/sun/git/AFLplusplus'
make -C utils/libdislocator
make[1]: Entering directory '/home/sun/git/AFLplusplus/utils/libdislocator'
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -I ../../include/ -Wall -g -Wno-pointer-sign  -shared -fPIC libdislocator.so.c -o libdislocator.so 
cp -fv libdislocator.so ../../
'libdislocator.so' -> '../../libdislocator.so'
make[1]: Leaving directory '/home/sun/git/AFLplusplus/utils/libdislocator'
make -C utils/libtokencap
make[1]: Entering directory '/home/sun/git/AFLplusplus/utils/libtokencap'
cc -O3 -funroll-loops -D_FORTIFY_SOURCE=2 -I ../../include/ -Wall -g -Wno-pointer-sign  -shared -fPIC libtokencap.so.c -o libtokencap.so "-ldl"
In file included from libtokencap.so.c:33:
../../include/../types.h:1:8: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘/’ token
 include/types.h
        ^
make[1]: *** [Makefile:72: libtokencap.so] Error 1
make[1]: Leaving directory '/home/sun/git/AFLplusplus/utils/libtokencap'
make: *** [GNUmakefile:608: distrib] Error 2

**Comment by vanhauser-thc** [comment]
what is the output when you do `cc -v` ?

btw please use the dev branch, I pushed a fix making building libtokencap optional (which fails for you, and it is really optional, unlikely you will use it)
